mesonbuild/coredata.py

```diff
@@ -268,6 +268,7 @@ def __init__(self, options):
         self.compilers = OrderedDict()
         self.cross_compilers = OrderedDict()
         self.deps = OrderedDict()
+        self.compiler_check_cache = OrderedDict()
```

Comment:
Can you annotate the keys to this with `# : ...`? Ideally we store everything that must be looked up from e.g. compiler options so we don't have to explicitly invalidate (bug prone) but just naturally get a cache miss if the key doesn't match.

Reply:
The key is defined in `Compiler.compile` (`compilers.py:1138`). I am personally against duplicating such things across multiple files, especially because there won't be a unit test for this comment. Would you be OK with a comment pointing to the function in question?

Reply:
Oh silly me, I hadn't realized that we were already caching in the class, so there was already a caching strategy etc. Sorry!